lib/deep-speech-interface.js

```diff
@@ -84,68 +84,80 @@ class DeepSpeechInterface {
       fs.mkdirSync(this._modelsDir);
     }
 
-    const localLMTxt = path.join(this._modelsDir, 'local_lm.txt');
-    const localLMArpa = path.join(this._modelsDir, 'local_lm.arpa');
-    const localLMBinary = path.join(this._modelsDir, 'local_lm.binary');
-    fs.writeFileSync(localLMTxt, finalGrammar.join('\n'));
+    const lmText = path.join(this._modelsDir, 'lm.txt');
+    fs.writeFileSync(lmText, finalGrammar.join('\n'));
 
-    const lmplz = spawnSync(
-      path.join(this._binDir, 'lmplz'),
+    const generateLm = spawnSync(
+      'python3',
       [
-        '--memory', '64M',
-        '--order', '2',
+        path.join(this._binDir, 'generate_lm.py'),
+        '--input_txt', lmText,
+        '--output_dir', this._modelsDir,
+        '--top_k', '500000',
+        '--kenlm_bins', this._binDir,
+        '--arpa_order', '5',
+        '--max_arpa_memory', '50%',
+        '--arpa_prune', '0|0|1',
+        '--binary_a_bits', '255',
+        '--binary_q_bits', '8',
+        '--binary_type', 'trie',
         '--discount_fallback',
-        '--text', localLMTxt,
-        '--arpa', localLMArpa,
       ]
     );
 
     if (this._debug) {
-      console.debug('lmplz: status:', lmplz.status);
-      console.debug('lmplz: stdout:', lmplz.stdout.toString());
-      console.debug('lmplz: stderr:', lmplz.stderr.toString());
+      console.debug('generate_lm.py: status:', generateLm.status);
+      console.debug('generate_lm.py: stdout:', generateLm.stdout.toString());
+      console.debug('generate_lm.py: stderr:', generateLm.stderr.toString());
     }
 
-    const buildBinary = spawnSync(
-      path.join(this._binDir, 'build_binary'),
+    const generateScorer = spawnSync(
+      path.join(this._binDir, 'generate_scorer_package'),
       [
-        '-a', '255',
-        '-q', '8',
-        'trie',
-        localLMArpa,
-        localLMBinary,
+        '--lm', path.join(this._modelsDir, 'lm.binary'),
```

Comment:
Not exactly sure what `--default_alpha` and `--default_beta` should be. Does it need to match what we use below for `this._model.setScorerAlphaBeta()`?

Reply:
No, those are the one we write in the model as default value. The setScorerAlphaBeta allows to use other value at runtime

Reply:
Ok. Before we were using `0.75` and `1.85`. The values I've used here were just taken from the DS docs. Any suggestion as to which I should use?

Reply:
That depends on your dataset / problem.

Reply:
I switched back to the old values, will evaluate later.